### PR TITLE
Generic WgslVec. Use associated const/type for WgslPrimitive

### PR DESCRIPTION
**Pull request by ivarflakstad**

### CODE REVIEW

File: crates/ratchet-core/src/ops/gemv.rs [hunk 1]
```diff
@@ -159,14 +159,15 @@ impl GEMV {
             let batchB = batch % metadata.bShape.x;
         });
 
+        let n = P::W;
```
Comment (ivarflakstad) [inline crates/ratchet-core/src/ops/gemv.rs:162]: If `wgsl!` supported `X::Y` patterns we could reference it directly.

File: crates/ratchet-core/src/ops/softmax.rs [hunk 2]
```diff
@@ -99,14 +103,16 @@ impl Softmax {
...
+        self.register_bindings::<P>(&mut kernel_builder, inplace)?;
         kernel_builder.write_metadata::<SoftmaxMeta>();
 
-        let dt = T::DT;
+        let dt = P::T::DT;
         let accessor = P::render_type();
 
         let BLOCK_SIZE = workgroup_size.x.render();
-        let minFloat = T::from(-65500).unwrap().render();
+        let minFloat = <<P as WgslPrimitive>::T as NumCast>::from(-65500)
```
Comment (ivarflakstad) [inline crates/ratchet-core/src/ops/softmax.rs:113]: There is probably a less verbose way of doing this.
All the float types could have the minimum value as an associated const (would not be surprised if that is already the case)

Comment (FL33TW00D) [reply crates/ratchet-core/src/ops/softmax.rs:113]: @ivarflakstad They do have a minimum value, unfortunately the rust minimum value for rust is slightly lower than the minimum value for WGSL.

Comment (ivarflakstad) [reply crates/ratchet-core/src/ops/softmax.rs:113]: Set your own as an associated const?

Comment (ivarflakstad) [reply crates/ratchet-core/src/ops/softmax.rs:113]:
```rust
trait WgslDType {
  const MIN: usize
}

impl WgslDType for f32 {
  const MIN: usize = 42
}
```
or something like that

**Pull request merged**